staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config_test.go

```diff
@@ -646,6 +648,187 @@ func TestKMSPluginHealthz(t *testing.T) {
 	}
 }
 
+// tests for masking rules
+func TestWildcardMasking(t *testing.T) {
```

Comment:
Add more complicated test cases with multiple `apiserverconfig.ResourceConfiguration`s.  This currently only tests overlap within a single `apiserverconfig.ResourceConfiguration` and not across multiple `apiserverconfig.ResourceConfiguration`s.